infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -469,6 +483,7 @@ private void finishAllStartupTracesAsFailures(Throwable reason) {
                     namespace,
                     startSynchronizer)
                 .bootstrapAsync();
+        bootstrapperFuture.whenComplete((res, ex) -> tracerUtil.finishSpan(tracingSpan));
```

Comment:
Should not you check if an ex is null here?

Reply:
Yep, done.